cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/RemoteTaskExecutor.java

```diff
@@ -141,9 +142,15 @@ public byte[] runTask(RunnableTaskRequest runnableTaskRequest) throws Exception
         }
       }, retryStrategy, Retries.DEFAULT_PREDICATE);
     } catch (Exception e) {
+      Exception ex = e;
+      if (e instanceof ServiceException) {
+        BasicThrowable basicThrowable = GSON
+          .fromJson(e.getMessage(), BasicThrowable.class);
```

Comment:
This is pretty fragile as you even don't know the original content-type. I am wondering if we should just do it inside the RemoteClient. Just parse it over there and add it as a cause for ServiceException. Then you can unwrap it here if you wish. 
Or if it's specific to `RemoteTaskExecutor`, add original content-type and json as separate fields to `ServiceException` or it's child and use it here. And be sure to handle any JSON deserialization problems in either case

Reply:
It's specific to `RemoteTaskExecutor` so I have added content-type as a field in `ServiceException`